### PR TITLE
Truncate invocation patterns based on char count

### PR DESCRIPTION
**Pull request by bduffany**
* Always show the first pattern
* Beyond that, truncate based on char count instead of target count

---

**Version bump**: Patch 

**Related issues**: Fixes https://github.com/buildbuddy-io/buildbuddy-internal/issues/664

### CODE REVIEW

File: server/util/invocationutil/invocationutil.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package invocationutil
```
Comment (tylerwilliams) [inline server/util/invocationutil/invocationutil.go:1]: Awesome that you factored this out!

Could we call it something different than util/invocationutil tho? It stutters (util), and also i don't think it's broadly useful.

How about we put a package called "format" or "invocation_format" inside of server/build_event_protocol?

File: server/util/invocationutil/invocationutil.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	patternCharLimit = 50
+	listSeparator    = ", "
```
Comment (tylerwilliams) [inline server/util/invocationutil/invocationutil.go:10]: Please comment these so that someone else could figure out how to set them.

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: server/util/invocationutil/invocationutil.go (hunk 1 above)
Comment (bduffany) [reply server/util/invocationutil/invocationutil.go]: sg, I wasn't happy with the name invocationutil either :P moved to `server/build_event_protocol/invocation_format`

File: server/util/invocationutil/invocationutil.go (hunk 2 above)
Comment (bduffany) [reply server/util/invocationutil/invocationutil.go]: Done

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**